### PR TITLE
Graceful BackupEntry deletion for specific shoot purposes

### LINKED ISSUE #3630: Retain backups only for deleted productive clusters
**How to categorize this issue?**
/area backup
/kind enhancement
/priority normal

**What would you like to be added**:
Let's retain only backups for deleted productive clusters, but not clusters with other `purpose` labels. We could then also increase the retention period for deleted productive clusters (e.g. 14 days).

**Why is this needed**:
We see applications of Gardener with a churn rate of 1000s of clusters per day leaving tens of thousands of backups per week in the blob store.

/cc @rfranzke @shreyas-s-rao

### PR DESCRIPTION
**Pull request by rfranzke**
**How to categorize this PR?**

/area backup
/kind enhancement
/priority normal

**What this PR does / why we need it**:
This PR allows to configure for which shoot purposes the `BackupEntry` deletion grace period applies.

**Which issue(s) this PR fixes**:
Fixes #3630

**Release note**:

```feature operator
It is now configurable for which shoot purposes the `BackupEntry` deletion grace period applies. An empty list (default) means that it applies for all shoot purposes (as it was earlier). If you want to only select specific purposes then please configure `.controllers.backupEntry.deletionGracePeriodShootPurposes[]` in the gardenlet's component configuration.
```

### CODE REVIEW

**Review by timebertt — DISMISSED** [review]
Looks great!
Would this be a good opportunity now to add some documentation for the BE controller in the gardenlet to https://github.com/gardener/gardener/blob/master/docs/concepts/gardenlet.md?
/lgtm

**Comment by rfranzke** [comment]
@timebertt I've added some docs now and also added some validation for the specified shoot purposes in the gardenlet configuration.

/squash

File: docs/concepts/gardenlet.md [hunk 1]
```diff
@@ -180,6 +180,35 @@ is available.
...
+The gardenlet consists out of several controllers which are now described in more detail.
+
+⚠️ This section is not necessarily complete and might be under construction.
+
+### `BackupEntry` Controller
+
+The `BackupEntry` controller reconciles those `core.gardener.cloud/v1beta1.BackupEntry` resources whose `.spec.seedName` value is equal to the name of a `Seed` the respective gardenlet is responsible for.
+Those resources are created by the `Shoot` controller (only if backup is enabled for the respective `Seed`) and there is exactly one `BackupEntry` per `Shoot`.
+
+The controller creates an `extensions.gardener.cloud/v1alpha1.BackupEntry` resource (non-namespaced) in the seed cluster and waits until the responsible extension controller reconciled it (see [this](../extensions/backupentry.md) for more details).
```
Comment (timebertt) [inline docs/concepts/gardenlet.md:194]: /cc @BeckerMax 
as you were also interested in documentation for this.

**Review by timebertt — APPROVED** [review]
Very nice!
Thanks for adding the documentation.
/lgtm

**Review by ialidzhikov — APPROVED** [review]
/lgtm

**Pull request merged**